services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/BatchOverrideConfiguration.java

```diff
@@ -0,0 +1,449 @@
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Configuration values for the BatchManager Implementation.  All values are optional, and the default values will be used if they
+ * are not specified.
+ */
+@SdkPublicApi
+public final class BatchOverrideConfiguration implements ToCopyableBuilder<BatchOverrideConfiguration.Builder,
+    BatchOverrideConfiguration> {
+
+    private final Integer maxBatchItems;
+    private final Integer maxBatchKeys;
+    private final Integer maxBufferSize;
+    private final Duration maxBatchOpenDuration;
+    private final Duration visibilityTimeout;
+    private final Duration longPollWaitTimeout;
+    private final Duration minReceiveWaitTime;
+    private final Integer maxDoneReceiveBatches;
+    private final List<MessageSystemAttributeName> messageSystemAttributeNames;
+    private final List<String> receiveMessageAttributeNames;
+    private final Boolean adaptivePrefetching;
+    private final Integer maxInflightReceiveBatches;
+
+    public BatchOverrideConfiguration(Builder builder) {
+        this.maxBatchItems = Validate.isPositiveOrNull(builder.maxBatchItems, "maxBatchItems");
+        this.maxBatchKeys = Validate.isPositiveOrNull(builder.maxBatchKeys, "maxBatchKeys");
+        this.maxBufferSize = Validate.isPositiveOrNull(builder.maxBufferSize, "maxBufferSize");
+        this.maxBatchOpenDuration = Validate.isPositiveOrNull(builder.maxBatchOpenDuration, "maxBatchOpenDuration");
+        this.visibilityTimeout = Validate.isPositiveOrNull(builder.visibilityTimeout, "visibilityTimeout");
+        this.longPollWaitTimeout = Validate.isPositiveOrNull(builder.longPollWaitTimeout, "longPollWaitTimeout");
+        this.minReceiveWaitTime = Validate.isPositiveOrNull(builder.minReceiveWaitTime, "minReceiveWaitTime");
+        this.messageSystemAttributeNames = builder.messageSystemAttributeNames;
+        this.receiveMessageAttributeNames = builder.receiveMessageAttributeNames;
+        this.adaptivePrefetching = builder.adaptivePrefetching;
+        this.maxInflightReceiveBatches = builder.maxInflightReceiveBatches;
+        this.maxDoneReceiveBatches = builder.maxDoneReceiveBatches;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * @return the optional maximum number of messages that are batched together in a single request.
+     */
+    public Integer maxBatchItems() {
+        return maxBatchItems;
+    }
+
+    /**
+     * @return the optional maximum number of batchKeys to keep track of.
+     */
+    public Integer maxBatchKeys() {
+        return maxBatchKeys;
+    }
+
+    /**
+     * @return the maximum number of items to allow to be buffered for each batchKey.
+     */
+    public Integer maxBufferSize() {
+        return maxBufferSize;
+    }
```

Comment:
It's unclear how this differs from maxBatchItems.

Reply:
maxBatchItems ==> Its maximum number of requests that are bacthed , maxBatchItems=10 means maximum 10 request will be sent
maxBuffferSize is maximum messages that can be bufferred , if it exceeds beyond this and if the messages are not processed then we throw exception